app/controllers/benefits/ReceiveAnyBenefitsController.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.benefits
+
+import config.{AppConfig, ErrorHandler}
+import controllers.employment.routes.CheckYourBenefitsController
+import controllers.predicates.{AuthorisedAction, InYearAction}
+import forms.YesNoForm
+import models.User
+import models.employment.BenefitsViewModel
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.EmploymentSessionService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.benefits.ReceiveAnyBenefitsView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ReceiveAnyBenefitsController @Inject()(implicit val cc: MessagesControllerComponents,
+                                             authAction: AuthorisedAction,
+                                             inYearAction: InYearAction,
+                                             receiveAnyBenefitsView: ReceiveAnyBenefitsView,
+                                             appConfig: AppConfig,
+                                             employmentSessionService: EmploymentSessionService,
+                                             errorHandler: ErrorHandler,
+                                             ec: ExecutionContext,
+                                             clock: Clock) extends FrontendController(cc) with I18nSupport with SessionHelper {
+
+  def yesNoForm(implicit user: User[_]): Form[Boolean] = YesNoForm.yesNoForm(
+    missingInputError = s"receiveAnyBenefits.errors.noRadioSelected.${if (user.isAgent) "agent" else "individual"}"
+  )
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+      employmentSessionService.getSessionData(taxYear, employmentId).map {
+        case Some(cya) =>
+          cya.employment.employmentBenefits match {
+            case Some(model) => Ok(receiveAnyBenefitsView(yesNoForm.fill(model.isBenefitsReceived), taxYear, employmentId))
+            case None => Ok(receiveAnyBenefitsView(yesNoForm, taxYear, employmentId))
+          }
+        case None =>
+          Redirect(CheckYourBenefitsController.show(taxYear, employmentId))
```

Comment:
can redirect straight to the overview page as the benefits CYA controller redirects you there anyway (line 59)

Reply:
updated